notify_slack.rb

```diff
@@ -0,0 +1,68 @@
+require 'uri'
+require 'net/http'
+require 'json'
+
+set :user_name, `whoami`.chomp
+ 
+def _send_message(message, channel)
+  payload = {
+    :text => message,
+    :channel => channel,
+    :username => 'Capiche Kun',
+    :icon_url => '',
+    :icon_emoji => ':capiche:',
+  }
+ 
+  uri = URI.parse(fetch(:webhook_url))
+ 
+  response = nil
+ 
+  request = Net::HTTP::Post.new(uri.request_uri)
+  request.set_form_data({ :payload => JSON.generate( payload ) })
+ 
+  http = Net::HTTP.new(uri.host, uri.port)
+  http.use_ssl = true
+  http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+ 
+  http.start do |h|
+    response = h.request(request)
+  end
+ 
+  response
+end
+
+def _slack_channel
+  (fetch(:chef_environment) == "testing") ? "dev_test" : "#dev-deploy"
+end
+
+
+namespace :notify do
```

Comment:
The namespace :slack is better idea.
I would like to distinguish deployment start from its finish. So I need one more nest, such as slack:deploy:start and  slack:deploy:finish.